<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Sudarshan A. Shetty" />


<title>Metagenomics workflow using RIVM-Toolbox</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">RIVM-Workflows</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="01_RIVM-ToolBoxCaseStudy.html">Case Study</a>
</li>
<li>
  <a href="02_Metagenomics_Humann3.html">WGS Metagenomics Humann3</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Metagenomics workflow using RIVM-Toolbox</h1>
<h3 class="subtitle">Humann3 pipeline</h3>
<h4 class="author">Sudarshan A. Shetty</h4>
<h4 class="date">Tue Nov 16 17:36:44 2021</h4>

</div>


<p>In this workflow, we demonstrate the use of Humann3 pipeline setup at RIVM using Singularity for Whole Genome ShotGun metagenomics microbiome data. This is part of the RIVM-toolbox. The snakemake workflow from raw reads to taxonomic and functional profiles was developed by Jeroen Frank from Bioinformatics department at RIVM.<br />
The source code is located here: <a href="https://gitlab.rivm.nl/frankj/humann_pipeline" class="uri">https://gitlab.rivm.nl/frankj/humann_pipeline</a><br />
A copy is forked to the RIVM-Microbiome GitLab account here: <a href="https://gitlab.rivm.nl/rivm-microbiome/humann_pipeline" class="uri">https://gitlab.rivm.nl/rivm-microbiome/humann_pipeline</a></p>
<p>First create a conda environment. This is a virtual environment which allows for convinient installation and management of different softwares. For some introduction check this <a href="https://towardsdatascience.com/virtual-environments-for-absolute-beginners-what-is-it-and-how-to-create-one-examples-a48da8982d4b">blog</a>. More information on conda can be found on this <a href="https://docs.conda.io/projects/conda/en/latest/">website</a>.<br />
Conda is installed on individual servers at RIVM. You can login to <code>Wetenschappelijke werkplek</code>.</p>
<div id="conda" class="section level1">
<h1>Conda</h1>
<div id="create-conda-env" class="section level2">
<h2>Create Conda Env</h2>
<pre class="bash"><code>
# Humann3 pipeline from frankj
conda create --name humann3_frankj -c bioconda snakemake 
# activate the environment. 
</code></pre>
<p>Before the environment is activated your terminal will start like this:<br />
<code>(base) [shettys@rivm-biohn-l05p humann_pipeline-master]$</code></p>
</div>
<div id="activate-conda-env" class="section level2">
<h2>Activate Conda Env</h2>
<pre class="bash"><code>
conda activate humann3_frankj
</code></pre>
<p>After the :<br />
<code>(humann3_frankj) [shettys@rivm-biohn-l05p humann_pipeline-master]$</code></p>
</div>
</div>
<div id="project-setup" class="section level1">
<h1>Project setup</h1>
<pre class="bash"><code># got to your working directory
cd /data/BioGrid/username/projects
# make a new directory
mkdir my_wgs_project
cd my_wgs_project</code></pre>
<p>Now download the raw files in a folder called <code>my_input_fqs</code> within the <code>my_wgs_project</code> folder. Also download the humann_pipeline from here <a href="https://gitlab.rivm.nl/frankj/humann_pipeline" class="uri">https://gitlab.rivm.nl/frankj/humann_pipeline</a> and unzip it.</p>
<pre class="bash"><code># The folder structure it something like this
my_wgs_project

    -- my_input_fqs\
            -- myR1.fastq.gz
            -- myR2.fastq.gz
            
    -- humann_pipeline-master\
            -- config  # Config file where parameters can be changed
            -- data  # Not applicable here
            -- envs # Software environment
            -- humann.py  # pipeline main function
            -- log.txt # Logs
            -- README.md # info
            -- run_pipeline.sh # internal function
            -- scripts # internal function
            -- smk_profiles # internal function
            -- snakefiles # internal function
</code></pre>
<p>Go to the <code>humann_pipeline-master</code> folder.</p>
<pre class="bash"><code>cd humann_pipeline-master/</code></pre>
<p>We can now submit our analysis (job) to the HPC with following command.</p>
</div>
<div id="run-pipeline-hpc" class="section level1">
<h1>Run Pipeline HPC</h1>
<pre class="bash"><code>
bsub -J mytestWGS -q bio -n 1 -o log.txt python humann.py --fq_dir ../my_input_fqs/ --out_dir ../my_input_fqs/mytestWGS_Output

# Note: 
# HPC LSF specific
# -J name of the job
# -q use bio cluster. There is also bio-prio cluster for priority jobs
# -n 1 use One core 
# -o output log of cluster job

# Pipeline specific 
# --fq_dir is the location of fastq.gz files here it is ../my_input_fqs/ that is one folder outside of humann_pipeline-master 
# --out_dir is the location for storing all outputs from the pipeline</code></pre>
<p>More information on HPC at RIVM: <a href="https://gitlab.rivm.nl/bioinformatics/docs/wikis/HPC/hosts_and_queues" class="uri">https://gitlab.rivm.nl/bioinformatics/docs/wikis/HPC/hosts_and_queues</a><br />
<a href="https://gitlab.rivm.nl/bioinformatics/docs" class="uri">https://gitlab.rivm.nl/bioinformatics/docs</a></p>
<p>Software tools used in this pipeline:</p>
<p><strong>Introduction</strong><br />
The HUMAnN 3.0 pipeline performs the following tasks:<br />
- Sequence data quality control reports - <a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a><br />
- Read length and quality filtering, adapter trimming - <a href="https://jgi.doe.gov/data-and-tools/bbtools/bb-tools-user-guide/bbduk-guide/">BBMapBBDuk</a><br />
- Contaminant (human) sequence removal - <a href="https://huttenhower.sph.harvard.edu/kneaddata/">KneadData</a><br />
- Estimate microbial composition - <a href="https://huttenhower.sph.harvard.edu/metaphlan/">MetaPhlAn 3.0</a><br />
- Estimate microbial metabolic pathway abundance - <a href="https://huttenhower.sph.harvard.edu/human">HUMAnN</a></p>
<p><strong>Input data</strong><br />
A folder containing paired-end metagenome sequencing data.</p>
<p><strong>Output data</strong> The main results are stored in the <code>results/</code> folder and include:</p>
<ul>
<li>metaphlan_bugs_list_merged.tsv<br />
</li>
<li><a href="https://github.com/biobakery/humann#1-gene-families-file">humann_genefamilies_KO_cpm_join.tsv</a><br />
converted uniref90 to KO, normalised counts to cpm, joined tables<br />
</li>
<li><a href="https://github.com/biobakery/humann#2-pathway-abundance-file">humann_pathabundance_cpm_join.tsv</a><br />
normalised counts to cpm, joined tables<br />
</li>
<li><a href="https://github.com/biobakery/humann#3-pathway-coverage-file">humann_pathcoverage_join.tsv</a><br />
Joined tables</li>
</ul>
<p>Intermediate and temporary data are automatically removed.</p>
<p>The output folder will have following folder</p>
<pre class="bash"><code>my_wgs_project

    -- my_input_fqs\
            -- myR1.fastq.gz
            -- myR2.fastq.gz
            -- mytestWGS_Output
                      -- fastqc 
                            -- pre_qc
                            -- post-qc
                      -- fq_input  
                            -- pseudo links to my_input_fqs
                      -- humann  
                            -- per sample humann output
                      --  kneaddata  
                      -- logs 
                      -- results
                            -- humann_genefamilies_KO_cpm_join.tsv  
                            -- humann_pathabundance_cpm_join.tsv  
                            -- humann_pathcoverage_join.tsv
                            -- humann_genefamilies_KO_join.tsv 
                            -- humann_pathabundance_join.tsv
                            -- metaphlan_bugs_list_merged.tsv
            </code></pre>
<p>The <code>metaphlan_bugs_list_merged.tsv</code> file can be read into R using the <code>biomeUtils::readMergedMetaphlan()</code> function.</p>
</div>
<div id="hand-off-to-rivm-toolbox" class="section level1">
<h1>Hand-off to RIVM-toolbox</h1>
<pre class="r"><code>path_to_file = &quot;my_wgs_project/my_input_fqs/mytestWGS_Output/results/metaphlan_bugs_list_merged.tsv&quot;
mpa.ps.lab8 &lt;- readMergedMetaphlan(input_file_path = path_to_file,
                              find_sample_name_pattern = &quot;_kneaddata_concat_metaphlan_bugs_list&quot;,
                              replace_sample_name_pattern = &quot;&quot;)</code></pre>
<p><strong>For more information contact:</strong><br />
Sudarshan A. Shetty: sudarshan.shetty[@]rivm[dot]nl<br />
Susana Fuentes: susana.fuentes[@]rivm[dot]nl</p>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
